runme.bh

```diff
@@ -2,24 +2,64 @@
 
 intro() {
     echo "Thank you for forking CodeZila"
-    echo "Please input any names in Upper Cammel Case and without any special characters"
-    read -p "What is the name of the algorithm or design you want to implement? (one word) " algorithm
-    checkInput $algorithm    
-    read -p "What language do you want to implement $algorithm in? " language
-    checkInput $language
-    createDir $algorithm $language
-    printTree $algorithm $language
-    checkDir $algorithm
+    echo "Please input any names without any special characters"
+    echo "For example: 'Insertion Sort' or 'Linked Lists'"
+    read -p "What is the name of the algorithm or design you want to implement? " algorithm
+    checkInput "$algorithm"    
+    pickLanguague "$algorithm"
+    printTree "$algorithm" "$language"
+    checkDir "$algorithm"
 }
 
 # Checks the input to see if it does not have illegal characters
 checkInput() {
     # $1 is the string being checked
-    if [[ $1 == *['!'@#\$%^\&*()_+]* ]]
+    if [[ "$1" == *['!'@\$%^\&*()_+]* ]]
     then
       echo "Illegal character used"
       exit
     fi
+    # Will capitalise everything after a space
+}
+
+# Asks the user that language they would like to use
+pickLanguague() {
+    echo "Select the language you want to implement "$1" in?"
+    echo "1)  C++ (.cpp)"
+    echo "2)  C++ (.c++)"
```

Comment:
Does anyone use this? For consistency let's only keep 1

Reply:
Not sure. I usually use cpp, but if you look at codezila/Graphs both c++ and cpp is used.

Reply:
I meant the extension (.c++). Noone uses it I think. You are right about folders. We can select either C++ or Cpp. C++ seems more clear to me. What about you?

Reply:
Make it C++ (.cpp)
C++ is the folder name
.cpp is the extension

Reply:
Sure, sounds good to me